### PR TITLE
feat: Global Project UI

### LINKED ISSUE #4416: Global (inherited) project settings
# Summary
Need support of global namespace allow list. Each project still has its own allow list. The final allow list is the concatenation of these two.

What change you think needs making.

# Motivation
Global namespace allow list eliminates the duplication of allow list for each project if they are the same.

Please give examples of your use case, e.g. when would you use this.

# Proposal
Configure global namespace allow list

How do you think this should be implemented?
1. In argocd-cm, add the following to specify global namespace allow list.
```
data: 
  projectSpec: |
    allowedResources:
     - apps/daemonsets
     - apps/replicasets
```

Or

```
data: 
  project.namespaceResourceWhitelist: |
  - group: ''
    kind: ResourceQuota
  - group: ''
    kind: LimitRange
  - group: ''
    kind: NetworkPolicy
```
2. When checking permission `IsGroupKindPermitted`, use the concatenation of allow list in project and and in argocd-cm.

### PR DESCRIPTION
**Pull request by mayzhang2000**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

fixes: #4416

### CODE REVIEW

**Comment by mayzhang2000** [comment]
image

**Review by alexmt — COMMENTED** [review]
added few comments

File: ui/src/app/shared/services/projects-service.ts [hunk 1]
```diff
@@ -103,6 +103,10 @@ export class ProjectsService {
         return requests.get(`/projects/${name}`).then(res => res.body as models.Project);
     }
 
+    public getGlobalProjects(name: string): Promise<models.GlobalProjectState> {
+        return requests.get(`/projects/${name}/globalprojects`).then(res => res.body as models.GlobalProjectState);
```
Comment (alexmt) [inline ui/src/app/shared/services/projects-service.ts:107]: I think it not worth defining a separate interface for `GlobalProjectState` . Please remove GlobalProjectState and just use:

```
public getGlobalProjects(name: string): Promise<models.Project[]> {
        return requests.get(`/projects/${name}/globalprojects`).then(res => res.body.globalprojects as models.Project[]);
}
```

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 2]
```diff
@@ -455,616 +455,790 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                            />
+
+                            <div className='white-box editable-panel'>
+                                <div className='white-box__details'>
+                                    <p>
+                                        NAMESPACE RESOURCE DENY LIST FROM GLOBAL PROJECT
+                                        {helpTip('Namespace-scoped K8s API Groups and Kinds which are prohibited from being deployed from global project')}
+                                    </p>
+                                    <React.Fragment>
+                                        {gpNamespaceResourceBlacklist.length > 0 ? (
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:1036]: Let's hide this section is it has no rows. And same for other inherited lists from global project.

**Review by alexmt — CHANGES_REQUESTED** [review]

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**